### PR TITLE
Feat/plpfilters

### PR DESCRIPTION
**Pull request by aka-sacci**
## What is this contribution about?

PLP filters added. Now, is possible to fillter the PLP products by filters in url. Should follow the pattern                                         `?filters.$FilterName=FilterValue`

## How to test it?

Use [product/productListiningPage.ts](https://deco.cx/admin/sites/otica-isabela/blocks/?returnLabel=Blocos&returnUrl=%2Fadmin%2Fsites%2Fotica-isabela%2Flibrary&ref=%23%2Fdefinitions%2FZGVjby1zaXRlcy9vdGljYS1pc2FiZWxhL2xvYWRlcnMvcHJvZHVjdC9wcm9kdWN0TGlzdGluaW5nUGFnZS50cw%3D%3D&type=product) in Deco panel or access http://localhost:8000/mycategory and define some filters in querystring. Use the pattern `?filters.$FilterName=FilterValue`

## Extra info

Capturar

Capturar

### CODE REVIEW

**Comment by aka-sacci** [comment]
Now, i will commit the Adapted PLP for route ` /busca` fully. Be warned

**Comment by aka-sacci** [comment]
Now, the PLP loader works on routes `/busca` properly. Just use the querystring `termo` and have fun. BTW, it accepts the `sort` querystring too.

How to test it?
Use [product/productListiningPage.ts](https://deco.cx/admin/sites/otica-isabela/blocks/?returnLabel=Blocks&returnUrl=%2Fadmin%2Fsites%2Fotica-isabela%2Flibrary&ref=%23%2Fdefinitions%2FZGVjby1zaXRlcy9vdGljYS1pc2FiZWxhL2xvYWRlcnMvcHJvZHVjdC9wcm9kdWN0TGlzdGluaW5nUGFnZS50cw%3D%3D&type=product) with the `/busca` route. Add the querystring `termo` and voila

Capturar2

Capturar3

Capturar4

File: packs/loaders/productListiningPage.ts [hunk 1]
```diff
@@ -22,6 +48,47 @@ const loaders = async (
   const url = new URL(req.url);
   const path = paths(config!);
 
+  const isCategoryPage = url.pathname.split("/")[1] != "busca";
```
Comment (gustavo-vasconcellos) [inline packs/loaders/productListiningPage.ts:51]:
```suggestion
  const isCategoryPage = !url.pathname.includes("busca");
```

File: packs/loaders/productListiningPage.ts [hunk 2]
```diff
@@ -22,6 +48,47 @@ const loaders = async (
...
+  const products = await fetchAPI<ProductData>(
+    path.product.getProduct({
+      offset: 32,
+      ...pageParams.productApiProps,
+      ...getSearchParams(url),
+    }),
+    { method: "POST" },
+  );
+
+  if (products.produtos.length == 0) return null;
```
Comment (gustavo-vasconcellos) [inline packs/loaders/productListiningPage.ts:76]:
```suggestion
  if (!products.produtos.length) return null;
```

File: packs/types.ts [hunk 3]
```diff
@@ -168,7 +170,7 @@ export interface GetProductProps {
 
   /**
    * @title Dynamic filters
-   * @description Define dinamic filters for the query. Example: Aro Fechado,Retangular */
+   * @description Define dinamic filters for the query. Its not possible to use them with "Term" parameter in "productList" and "ProductListiningPage" Loaders */
```
Comment (gustavo-vasconcellos) [inline packs/types.ts:173]:
```suggestion
   * @description Define dinamic filters for the query. Its not possible to use them with "Term" parameter */
```

**Review by gustavo-vasconcellos — APPROVED** [review]

**Pull request merged**